snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -249,6 +250,32 @@ def is_clean(self, step):
         except errors.NoLatestStepError:
             return True
 
+    def is_outdated(self, step):
+        """Return true if the given step needs to be updated (no cleaning)."""
```

Comment:
Can you clarify this doc string (what is in between parens).

Reply:
Done, though mostly by directing the reader to `get_outdated_report()`.